src/Mentat.UI/Models/Assignment.cs

```diff
@@ -1,23 +1,106 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
+using System.IO;
 using System.Linq;
 using System.Web;
+using Mentat.UI.Controllers;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.EntityFrameworkCore;
 
 namespace Mentat.UI.Models
 {
+    public class FileExtensionCheckAttribute : ValidationAttribute
```

Comment:
This class should go in its own file. It also needs a unit test. That said, time is pretty short. Can you make the class private and swap its placement on this page with Assignment. 

The top class definition belongs to the public class that shares its name with the file name.

Reply:
Don't worry about this not a big deal.